File: frontend/src/types/event.ts
export type EventStatus = "Draft" | "Published" | "Cancelled" | "Archived";

export interface Event {
  id: string;
  title: string;
  description: string;
  location: string;
  startDate: string;
  endDate: string;
  organizerId: string;
  status: EventStatus;
  imageUrl: string | null;
  websiteUrl: string | null;
  maxAttendees: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface LineItemTemplate {
  name: string;
  netAmount: number;
  taxRateId: string;
  taxRateName: string;
  taxRatePercentage: number;
}

export interface TicketType {
  id: string;
  eventId: string;
  name: string;
  description: string;
  price: number;
  currency: string;
  availableQuantity: number;
  soldQuantity: number;
  maxPerOrder: number;
  showRemainingQuantity: boolean;
  saleStart: string | null;
  saleEnd: string | null;
  lineItems: LineItemTemplate[];
}

File: src/DevConfTicketing.Domain/Events/Event.cs
using System.ComponentModel;
using System.Text.Json.Serialization;

namespace DevConfTicketing.Domain.Events;

[Description("Represents a conference or developer event")]
public class Event
{
    [Description("Unique identifier of the event")]
    [JsonPropertyName("id")]
    public required string Id { get; init; }

    [Description("Title of the event")]
    [JsonPropertyName("title")]
    public required string Title { get; set; }

    [Description("Detailed description of the event")]
    [JsonPropertyName("description")]
    public required string Description { get; set; }

    [Description("Physical or virtual location of the event")]
    [JsonPropertyName("location")]
    public required string Location { get; set; }

    [Description("Date and time when the event starts")]
    [JsonPropertyName("startDate")]
    public required DateTimeOffset StartDate { get; set; }

    [Description("Date and time when the event ends")]
    [JsonPropertyName("endDate")]
    public required DateTimeOffset EndDate { get; set; }

    [Description("Identifier of the event organizer")]
    [JsonPropertyName("organizerId")]
    public required string OrganizerId { get; set; }

    [Description("Current status of the event")]
    [JsonPropertyName("status")]
    public EventStatus Status { get; set; } = EventStatus.Draft;

    [Description("URL of the event banner or logo image")]
    [JsonPropertyName("imageUrl")]
    public string? ImageUrl { get; set; }

    [Description("URL of the event website")]
    [JsonPropertyName("websiteUrl")]
    public string? WebsiteUrl { get; set; }

    [Description("Maximum number of attendees allowed")]
    [JsonPropertyName("maxAttendees")]
    public int MaxAttendees { get; set; }

    [Description("Timestamp when the event was created")]
    [JsonPropertyName("createdAt")]
    public DateTimeOffset CreatedAt { get; init; }

    [Description("Timestamp when the event was last updated")]
    [JsonPropertyName("updatedAt")]
    public DateTimeOffset UpdatedAt { get; set; }
}
